py_ecc/bls/api.py

```diff
@@ -109,3 +113,48 @@ def verify_multiple(pubkeys: Sequence[BLSPubkey],
         return final_exponentiation == FQ12.one()
     except (ValidationError, ValueError, AssertionError):
         return False
+
+
+def verify_multiple_multiple(
```

Comment:
Renaming ideas: `optimized_verify_multiple`, `fast_verify_multiple`...

Reply:
If it's faster why not just replace the existing implementation instead of maintaining two?

Also, if there's an existing one and we are keeping both around, seems prudent to test them against each other to ensure they have equal behavior (sorry if I missed the test that does this)

Reply:
The new function is an optimized version of doing multiple times of `verify_multiple`. The later verifies a single attestation and the former verifies many attestations in a block.

Reply:
I do desperately want to get rid of `verify_multiple_multiple` and rename it properly. Besides the name of the function, the function parameter `pubkeys_and_messages: Sequence[Tuple[Sequence[BLSPubkey], Sequence[Hash32]]]` is also confusing. Should we create some objects like 
```python
class SignedMessage:
    signature: BLSSignature
    public_keys: Sequence[BLSPubkey]
    message_hashes: Sequence[Hash32]
    def __init__(self, signature, public_keys, message_hashes):
        self.signature = signature
        if len(public_keys) != len(message_hashes)
            raise ValidationError()
        self.public_keys = public_keys
        self.message_hashes = message_hashes
```
Then have
```
def verify_multiple(signed_message: SignedMessage): 
def verify_multiple_multiple(signed_messages: Sequence[SignedMessage]): 
```

Reply:
@ChihChengLiang 

>I do desperately want to get rid of verify_multiple_multiple and rename it properly.

how about `batch_verify` ?

Reply:
Sounds good. We could also do "verify_multiple_aggregate_signatures" to match milagro's function name.